FeedStoreChallenge/Core Data/CoreDataFeedStore.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  CoreDataFeedStore.swift
+//  FeedStoreChallenge
+//
+//  Created by Wilmer Barrios on 23/01/21.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import Foundation
+import CoreData
+
+public final class CoreDataFeedStore: FeedStore {
+	
+	typealias StoredFeedData = (feed: [CDFeedImageItem], timestamp: Date)
+	
+	private let context: NSManagedObjectContext
+	
+	public init(contextBuilder: CDContextBuilder) throws {
+		switch contextBuilder.build() {
+		case .success(let context):
+			self.context = context
+		case .failure(let e):
+			throw e
+		}
+	}
+	
+	// MARK: FeedStore extension
+	
+	public func deleteCachedFeed(completion: @escaping DeletionCompletion) {
+		let context = self.context
+		context.perform {
+			let fetchRequest: NSFetchRequest<NSFetchRequestResult> = CDFeedImage.fetchRequest()
+			let batchDeleteRequest = NSBatchDeleteRequest(fetchRequest: fetchRequest)
+			do {
+				try context.execute(batchDeleteRequest)
+				completion(nil)
+			} catch let e {
+				completion(e)
+			}
+		}
+	}
+	
+	public func insert(_ feed: [LocalFeedImage], timestamp: Date, completion: @escaping InsertionCompletion) {
+		let context = self.context
+		context.perform {
+			let feed_data: [CDFeedImageItem] = feed.toCD(context: context)
```

Comment:
The compiler can infer the type:

```suggestion
			let feed_data = feed.toCD(context: context)
```

Reply:
Thanks @caiozullo for all this feedback!

I have a question for this point, I've read about better compile times (the app I'm working on, has 450 - 500 seconds on clean build) and one suggestion was to explicitly write properties/variables types. Is this suggestion very beneficial? or doesn't worth it? What would be your suggestion?. Making MacOS modules seems really fast for large applications with high compile times. Thanks again!

https://medium.com/rocket-fuel/optimizing-build-times-in-swift-4-dc493b1cc5f5

Reply:
It's beneficial to add the types if the compiler takes a long time to infer it (e.g., complex syntax and chained closures are usually slow). My suggestion is to measure before optimizing - you can use build settings to find the slow places. Moreover, new Swift versions are getting better at it ✅

Reply:
Thanks for your feedback!

I'll be checking build times measure methods, have a good day!